### DISCUSSION TITLE
Face Restore Codeformer / GPFGAN slow speed compared to A1111

### DISCUSSION THREAD

**Discussion by TheRealPhooney** [lllyasviel/stable-diffusion-webui-forge#311] [General]
I couldn't see anybody else talking about this, so I wanted to check if it was just me.

Whenever I use face restore, either as part of txt2img/img2img or within the Reactor extension, the face restore part seems to take a lot longer than it did on A1111.

A quick and dirty comparison is a 512x768 image taking 3-4 seconds without any face restoration, and 12-14 seconds with face restoration, so 9-11 seconds for the GPFGAN/Codeformer to do its thing.

It's especially noticeable if you were using AnimateDiff, resulting in a bunch of frames to process.

Unfortunately, I pretty much gutted my A1111 installation moving all kinds of stuff over to Forge, and it would be a bit of a drag for me to set up A1111 again so all I have is my "feeling" that this part of the process is a lot slower rather than a direct comparison.

Is anybody else finding this slowdown too?

**Comment by huchenlei** [comment]
Verified. We had to disable face restore unittests in CQ as they just get stuck there.

**Comment by raj2318** [comment]
![Uploading 20240603_135457.jpg…]()

**Comment by rhshamim** [comment]
![Uploading 1724850325157.jpg…]()
